cabby/geo/visualize.py

```diff
@@ -0,0 +1,116 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Library to support geographical visualization.'''
+
+from typing import Tuple, Sequence, Optional, Dict, Text
+from json import JSONDecoder, JSONDecodeError
+import json
+import pandas as pd
+import folium
+import geopandas as gpd
+import shapely.geometry as geom
+from shapely.geometry import Polygon, Point, LineString
+
+
+from cabby.geo import util
+
+def read_file(path: Text) -> gpd.GeoDataFrame:
+    '''Read file.
+    Arguments:
+      path: path to file.
+    Returns:
+      The GeoDataFrame theat includes the start and end points, pivots and 
+      route.
+    '''
+
+    start = gpd.read_file(path, layer='start')
+    end = gpd.read_file(path, layer='end')
+    route = gpd.read_file(path, layer='route')
+    main = gpd.read_file(path, layer='main')
+    near = gpd.read_file(path, layer='near')
```

Comment:
Why do we have to read the file multiple times? Can we read it once and pull out what we need?

Reply:
Not possible according to this - https://stackoverflow.com/questions/56165069/can-geopandas-get-a-geopackages-or-other-vector-file-all-layers

In each line a different layer is read.

Reply:
Okay, no problem then!